src/js/components/devices/terminal.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from '@material-ui/core';
+
+import { XTerm } from 'xterm-for-react';
+import { FitAddon } from 'xterm-addon-fit';
+import { SearchAddon } from 'xterm-addon-search';
+import { encode, decode } from 'msgpack5';
+
+export const Terminal = props => {
+  const { deviceId } = props;
+  const [socket, setSocket] = useState(null);
+  const xtermRef = React.useRef(null);
+
+  const onData = data => {
+    const msg = { cmd: 'shell', data: data };
+    const encodedData = encode(msg);
+    socket.send(encodedData);
+  };
+
+  React.useEffect(() => {
+    const term = xtermRef.current.terminal;
+
+    fitAddon.fit();
+    term.resize(80, 40);
+
```

Comment:
I was trying this, and I had to (total hack) add here:
```js
document.cookie = 'JWT=' + jwt + '; path=/';
```
otherwise the cookie was not sent.

Reply:
Already managed here: https://github.com/mendersoftware/gui/pull/1047/commits/21a3fadbecab0a13929c9c833040c44e50c75d91